src/seedu/addressbook/Main.java

```diff
@@ -106,6 +106,11 @@ private CommandResult executeCommand(Command command)  {
         try {
             command.setData(addressBook, lastShownList);
             CommandResult result = command.execute();
+            
+            if(!storage.isFileExist()){
```

Comment:
this is not really the right place to check for the file presence. you should check that just before saving, in the `save()` method.

- in this approach, the error is thrown here, but never caught; which propagates to main and thrown to the user, and the execution halts.
- ideally we want to continue using the existing behavior of creating the file when it is not found, but here, we want to use the exception mechanism to let the user know of the error that happened.